timeboost-types/src/decryption.rs

```diff
@@ -189,22 +371,17 @@ impl DkgAccumulator {
         &self.bundles
     }
 
-    pub fn completed(&self) -> bool {
-        self.completed
-    }
-
     pub fn is_empty(&self) -> bool {
         self.bundles.is_empty()
     }
 
     pub fn try_add(&mut self, bundle: DkgBundle) -> Result<(), VessError> {
-        if self.completed {
-            return Ok(());
-        }
+        // caller should ensure that no bundles are added after finalization
         let aad: &[u8; 3] = b"dkg";
         let committee = self.store.committee();
-        let vess = Vess::new_fast_from(committee);
-        vess.verify(
+        let vess = Vess::new_fast();
+        vess.verify_shares(
```

Comment:
Unless performance has improved a lot I suggest to wrap this in a `tokio::task::block_in_place` so that the executor has a chance to move other tasks to different threads.